kafka/src/main/java/com/linecorp/armeria/server/logging/structured/kafka/ServiceLogJsonKafkaSerializer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured.kafka;
+
+import java.util.Map;
+
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class ServiceLogJsonKafkaSerializer<T> implements Serializer<T> {
```

Comment:
still doesn't work as the desired result type may different among underlying systems?
I mean, Kafka expects `byte[]` as the final form of any logs but different software such as logstash, fluented might expect `String` as the final form.